### ISSUE TITLE
Flag icons are broken

### ISSUE THREAD

**Issue by gornostal** [Ulauncher/ulauncher-emoji#14]
![image](https://user-images.githubusercontent.com/1202543/94999951-a0d5e280-05c5-11eb-9d60-e89bd496260d.png)
Although flag emojies are copied correctly to the clipboard

**Comment by gornostal** [comment]
@MyriaCore Any ideas where to look for a cause of this bug?

**Comment by MyriaCore** [comment]
Sorry for late reply. I'll start digging around

**Comment by gornostal** [comment]
No worries!

**Comment by MyriaCore** [comment]
![image](https://user-images.githubusercontent.com/35076430/95778215-a7cfb580-0c95-11eb-9a82-a71a012e49a2.png)
mine works, weird. I've done a ton of weird hacks to my stuff since tho, might have been something I changed. What emoji pack are u using?

**Comment by MyriaCore** [comment]
It appears to only affect noto and blobs:

> With noto:
>  
> ![image](https://user-images.githubusercontent.com/35076430/95778335-ec5b5100-0c95-11eb-8aa0-d67afe21dca6.png)
> 
> With blobs:
> 
> ![image](https://user-images.githubusercontent.com/35076430/95778378-0137e480-0c96-11eb-9c2a-0a72364f89dc.png)
> 
> With apple: 
> 
> ![image](https://user-images.githubusercontent.com/35076430/95778421-16147800-0c96-11eb-9249-1a1e45a5c7a8.png)

**Comment by gornostal** [comment]
`noto`

After merging I did some minor refactoring, but I also tested tested with the version from your branch and I saw the same issue there.

**Comment by gornostal** [comment]
> It appears to only affect noto and blobs:

Indeed.

**Comment by MyriaCore** [comment]
![image](https://user-images.githubusercontent.com/35076430/95778605-799ea580-0c96-11eb-8d11-6dee021330ce.png)

The icon entries for most of these flags are null in blobmoji / noto. Maybe a codepoint mismatch?

**Comment by gornostal** [comment]
Aah.. I had `noto` as a fallback theme too.
Changing it helped.

So I guess it's not a bug then. I'm going to close it

**Comment by gornostal** [comment]
Or you think it is a bug and the flags should be there with noto?

**Comment by MyriaCore** [comment]
Yeah, I *think* so. I don't think these flags are new, so it's probably not an emoji 13 thing, right? AFAIK noto and blob should support full emoji 11 (or maybe 12).

**Comment by gornostal** [comment]
I think I agree.
I also think that at this point you know much more about emojies than I do, so I'll trust your opinion. 
I haven't heard about noto or other emoji packs before your PR 😀

**Comment by MyriaCore** [comment]
[This](http://unicode.org/emoji/charts/emoji-list.html#1f1f1_1f1f9) is the unicode.org listing for the lithuania flag. The listed codepoint is `U+1F1F1 U+1F1F9`. 

Here's the function where we generate the URL for them:

https://github.com/Ulauncher/ulauncher-emoji/blob/d56b2c8106ab5fbc87b55f608a3bd75db23f2904/EmojiSpider.py#L68-L94

`base` for this codepoint should be `"1f1f1-1f1f9"`, so it probably attempts to retrieve the URL [here](https://github.com/C1710/blobmoji/raw/master/png/128/emoji_u1f1f1-1f1f9.png), which 404's.

**Comment by gornostal** [comment]
ah.. I see

**Comment by MyriaCore** [comment]
```shell
$ python -c 'print("🇱🇹".encode("unicode-escape"))'
b'\\U0001f1f1\\U0001f1f9'
```

So the codepoint seems correct. Maybe there's some schema that noto and blob is using that I didn't properly enforce.

**Comment by MyriaCore** [comment]
After looking at the [makefile](https://github.com/C1710/blobmoji/blob/437784e59a5dc524aca8b8ea520ea1d889d5643e/Makefile) for the blobmoji project, I've determined that the schema for the flags is actually `https://github.com/C1710/blobmoji/tree/master/third_party/region-flags/png/%s.png`, where `%s` is the [BCP 47](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) country/region/subregion code. See [the readme for the region-flags dir](https://github.com/C1710/blobmoji/tree/master/third_party/region-flags).

**Comment by MyriaCore** [comment]
Also in the [readme](https://github.com/C1710/blobmoji/blob/437784e59a5dc524aca8b8ea520ea1d889d5643e/README.md#color-emoji-assets) for blobmoji:

> Also note that the images in the font might differ from the original assets. In particular the flag images in the font are PNG images to which transforms have been applied to standardize the size and generate the wave and border shadow. We do not have SVG versions that reflect these transforms.

**Comment by MyriaCore** [comment]
There's a similar passage in the [readme](https://github.com/googlefonts/noto-emoji/blob/aac7ccaa4d1dea4543453b96f7d6fc47066a57ff/README.md#color-emoji-assets) for noto emoji. This is kinda worrying.

@gornostal the way I see it, we have 3 options:
1. Close the issue and cry. It looks like it'd be pretty difficult to add support for these emojis from this data source.
2. Try to see if their makefile can be modified to maintain these transformations in image form.
3. Find a different datasource for the noto and blob emoji set, like emojipedia. Related: #15
  - blobs: https://emojipedia.org/google/android-7.0/
  - noto: https://emojipedia.org/google/android-11.0/

**Comment by MyriaCore** [comment]
Wait NVM, I'm super wrong I think

**Comment by MyriaCore** [comment]
It looks like the make the name for the flag emojis [here](https://github.com/C1710/blobmoji/blob/437784e59a5dc524aca8b8ea520ea1d889d5643e/flag_glyph_name.py). The names are reformatted in 04x hex.  

Not sure why the normal URL of just:

```
https://github.com/C1710/blobmoji/raw/master/png/128/emoji_u1f1f1_1f1f9.png
``` 

doesn't work then.

**Comment by gornostal** [comment]
I think it's fine because there's a fallback theme. If it's set correctly the issue is gone.
Thanks a lot for looking into this!
